### DISCUSSION TITLE
Get ring with theme() function

### DISCUSSION THREAD

**Discussion by silveltman** [tailwindlabs/tailwindcss#11183] [Help]
Is it possible to get 'ring' properties with the theme() function? What I would like to do is the following:

```
addComponents({
	'.ringed': {
		boxShadow: theme('ring.1'),
		'--tw-ring-color': theme('colors.mycolor'),
                // ...
	}
})

**Comment by wongjn** [comment]
Yep! Access the property that you want:

https://github.com/tailwindlabs/tailwindcss/blob/cdbcefc232987c69141450951599169631332dde/stubs/config.full.js#L726-L749

**Reply by wongjn** [reply]
> Thanks for your quick reply @wongjn !
> 
> Does this mean we would have to manually construct the boxshadow property with something like this?
> 
> ```
> addComponents({
>   '.ringed': {
>     boxShadow: 'var(--tw-ring-inset) 0 0 0 calc(3px + var(--tw-ring-offset-width)) var(--tw-ring-color);'
>   }
> })
> ```
> 
> I would love to have more simplicity (like @apply) within the addComponent function. Being able to do something like this:
> 
> addComponents({ '.ringed': 'ring-1 ring-blue-500 ring-inset ...' })
> 
> I have not yet found a way to do it like this, nor have I found a plugin for this. So if there is, please let me know :)[^1]

You could consider using [`@apply`](https://tailwindcss.com/docs/reusing-styles#extracting-classes-with-apply), like:

```js
addComponents({
  '.ringed': {
    '@apply ring-1 ring-blue-500 ring-inset': {},
    // …
  },
});
```

[^1]: [Comment seemingly intended to be in this thread](https://github.com/tailwindlabs/tailwindcss/discussions/11183#discussioncomment-5840551)

**Reply by silveltman** [reply]
Seems a bit hacky, does this have any downsides?

**Reply by wongjn** [reply]
Yes, I would say the standout one would be not using Tailwind's utility classes directly. See [Tailwind's documentation on premature abstraction](https://tailwindcss.com/docs/reusing-styles#avoiding-premature-abstraction).

**Comment by silveltman** [comment]
Thanks for your quick reply @wongjn !

Does this mean we would have to manually construct the boxshadow property with something like this?
```
addComponents({
  '.ringed': {
    boxShadow: 'var(--tw-ring-inset) 0 0 0 calc(3px + var(--tw-ring-offset-width)) var(--tw-ring-color);'
  }
})
```

I would love to have more simplicity (like @apply) within the addComponent function. Being able to do something like this:

addComponents({
  '.ringed': 'ring-1 ring-blue-500 ring-inset ...'
})

I have not yet found a way to do it like this, nor have I found a plugin for this. So if there is, please let me know :)

**Reply by wongjn** [reply]
Responded at https://github.com/tailwindlabs/tailwindcss/discussions/11183#discussioncomment-5840675 to keep the reply chain together.
